packages/modules/src/dataSelection/DataSourceRow.tsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react'
+import { Utf8, Utf8Vector } from 'apache-arrow'
+import { RowLike } from 'apache-arrow/type'
+
+import TableRow from '@material-ui/core/TableRow'
+import TableCell from '@material-ui/core/TableCell'
+import Tooltip from '@material-ui/core/Tooltip'
+import Checkbox from '@material-ui/core/Checkbox'
+
+import { useDataRepositoryItemValue, DataRepositoryItemsTable } from '@dharpa-vre/client-core'
+import { LoadingIndicator } from '@dharpa-vre/client-ui'
+
+import useStyles from './DataSourceRow.styles'
+
+import { TableView } from '../components/TableView'
+
+type DataSourceRowProps = {
+  repositoryItemBatch: DataRepositoryItemsTable
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  repositoryItem: RowLike<any>
```

Comment:
You probably wanted to say `RowLike<{ id: string }>` because you know that there is an `id` field in this row when you access it as `selectedItemIds.includes(repositoryItem[0])` down below. I would not recommend using `any` - it's a time bomb.

Reply:
I actually wanted to say `RowLike<DataRepositoryItemStructure>`, but after wasting quite a bit of time, I still couldn't get this type right, and just gave up. TypeScript still complains with `RowLike<{ id: string }>`.

Reply:
No worries, leave the PR open and I fix the type later today or tomorrow.